### PR TITLE
Language Tags for the texts in profile\authCallback.php

### PR DESCRIPTION
**Pull request by RamisaZaman**
# Pull Request Checklist:
profile\authCallback.php English, Spanish, and French Translation.

# Pre-Approval

- [x] There is a description section in the pull request that details what the proposed changes do. It can be very brief if need be, but it ought to exist.
- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] All new text is preferably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages), and the [spreadsheet tracking internationalizations](https://docs.google.com/spreadsheets/d/133fps9w2pUCEjUA6IGCcQotk7dn9KvepMXJ2IWUZsE8/edit?usp=sharing) has been updated either with a new row or with checkmarks to existing rows.
- [x] There are no linter errors
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] If this PR makes any changes that would require additional configuration of any Symbiota portals outside of the files tracked in this repository, make sure that those changes are detailed in [this document](https://docs.google.com/document/d/1T7xbXEf2bjjm-PMrlXpUBa69aTMAIROPXVqJqa2ow_I/edit?usp=sharing).

# Post-Approval

- [ ] It is the code author's responsibility to merge their own pull request after it has been approved
- [ ] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [ ] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [ ] If this PR represents a hotfix into the `master` branch, a subsequent PR from `master` into `Development` should be made **merge** option (i.e., no squash).
- [ ] If the dev team has agreed that this PR represents the last PR going into the `Development` branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place. Follow the release protocol [here](https://github.com/BioKIC/Symbiota/blob/master/docs/release-protocol.md).
- [ ] Don't forget to delete your feature branch upon merge. Ignore this step as required.

Thanks for contributing and keeping it clean!

### CODE REVIEW

File: profile/authCallback.php [hunk 1]
```diff
@@ -25,15 +28,15 @@
     $status = $oidc->authenticate();
   }
   catch (Exception $ex){
-    $_SESSION['last_message'] = 'Caught exception: ' . $ex->getMessage() . ' <ERR/>';
-    header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+    $_SESSION['last_message'] = $LANG['CAUGHT_EXCEPTION'] . $ex->getMessage() . ' <ERR/>';
```
Comment (themerekat) [inline profile/authCallback.php]: It's generally best to not include spaces in the lang tags, as they're liable to get lost. Can you take it out of the lang tag and include it in the concatenated statement here?

Comment (themerekat) [reply profile/authCallback.php]: @Atticus29, I think these don't need lang tags, is that correct?

File: profile/authCallback.php [hunk 2]
```diff
@@ -25,15 +28,15 @@
...
+    header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (themerekat) [inline profile/authCallback.php]: @Atticus29 , I think this doesn't need a lang tag, correct? Ed always removes them from my internationalizations, but I've never asked

File: profile/authCallback.php [hunk 3]
```diff
@@ -25,15 +28,15 @@
...
     exit();
   }  
   if($status){
     $sub = $oidc->requestUserInfo('sub');
     if($profManager->authenticate($sub, $providerUrls['oid'])){
       if($_SESSION['refurl']){
-        header("Location:" . $_SESSION['refurl']);
+        header($LANG['LOCATION'] . $_SESSION['refurl']);
```
Comment (themerekat) [inline profile/authCallback.php]: See above

File: content/lang/profile/authCallback.en.php [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<?php
+/*
+------------------
+Language: English
+------------------
+*/
+
+$LANG['CAUGHT_EXCEPTION'] = 'Caught exception: ';
+$LANG['LOCATION'] = 'Location:';
+$LANG['UNKNOWN_ERROR'] = 'Unkown Error - Could not authenticate - try again later or alert a system admin';
```
Comment (themerekat) [inline content/lang/profile/authCallback.en.php]: Typo

**Comment by themerekat** [comment]
@RamisaZaman , good work on this! I want @Atticus29 to confirm whether all these strings need language tags, but if they do, it looks good, just a few comments.

**Review by themerekat — CHANGES_REQUESTED** [review]

File: profile/authCallback.php (hunk 2 above)
Comment (Atticus29) [reply profile/authCallback.php]: Correct. No lang tag needed here.

File: profile/authCallback.php (hunk 3 above)
Comment (Atticus29) [reply profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php [hunk 5]
```diff
@@ -43,34 +46,34 @@
         try{
           $status = $profManager->linkLocalUserOidSub($email, $sub, $oidc->getProviderURL());
         }catch (Exception $ex){
-          $_SESSION['last_message'] = 'Caught exception: ' . $ex->getMessage();
-          header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+          $_SESSION['last_message'] = $LANG['CAUGHT_EXCEPTION'] . $ex->getMessage();
+          header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php (hunk 1 above)
Comment (Atticus29) [reply profile/authCallback.php]: @themerekat , after thinking about what this is actually doing, I realized that these DO need to be translated. The 'last_message' session gets tracked and output to the end user in profile/index.php.

File: profile/authCallback.php [hunk 6]
```diff
@@ -43,34 +46,34 @@
...
           exit();
         }
         if($status){
           if($profManager->authenticate($sub, $providerUrls['oid'])){
             if($_SESSION['refurl']){
-              header("Location:" . $_SESSION['refurl']);
+              header($LANG['LOCATION'] . $_SESSION['refurl']);
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php [hunk 7]
```diff
@@ -43,34 +46,34 @@
...
               unset($_SESSION['refurl']);
             }
           }
           else{
-            $_SESSION['last_message'] = "Unkown Error - Could not authenticate - try again later or alert a system admin <ERR/>";
-            header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+            $_SESSION['last_message'] = $LANG['UNKNOWN_ERROR'] . " <ERR/>";
+            header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php [hunk 8]
```diff
@@ -43,34 +46,34 @@
...
             //@TODO Consider logging this error to PHP logfiles
           }
         }else{
-          $_SESSION['last_message'] = "Error - Could not authenticate with Authentication provider <ERR/>";
-          header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+          $_SESSION['last_message'] = $LANG['ERROR'] . " <ERR/>";
+          header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php [hunk 9]
```diff
@@ -43,34 +46,34 @@
...
         }
         
       }
       else{
-        $_SESSION['last_message'] = "Unable to retrieve email address from authentication provider. <ERR/>";
-        header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+        $_SESSION['last_message'] = $LANG['UNABLE_RETRIEVE_EMAIL'] . " <ERR/>";
+        header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: profile/authCallback.php [hunk 10]
```diff
@@ -43,34 +46,34 @@
...
       }
     }
   }
-  $_SESSION['last_message'] = "Authentication failed. <ERR/>";
-  header('Location:' . $CLIENT_ROOT . '/profile/index.php');
+  $_SESSION['last_message'] =  $LANG['AUTHENTICATION_FAILED'] . " <ERR/>";
+  header($LANG['LOCATION'] . $CLIENT_ROOT . '/profile/index.php');
```
Comment (Atticus29) [inline profile/authCallback.php]: No lang tag needed.

File: content/lang/profile/authCallback.en.php [hunk 11]
```diff
@@ -0,0 +1,15 @@
+<?php
+/*
+------------------
+Language: English
+------------------
+*/
+
+$LANG['CAUGHT_EXCEPTION'] = 'Caught exception: ';
+$LANG['LOCATION'] = 'Location:';
```
Comment (Atticus29) [inline content/lang/profile/authCallback.en.php]: I think that the location tag can be removed entirely from this file (and from the spanish and french versions as well).

**Review by Atticus29 — CHANGES_REQUESTED** [review]
A few minor changes requested, but good job and thank you!

**Review by Atticus29 — APPROVED** [review]
Looks great! Thank you!

**Review by themerekat — APPROVED** [review]

**Pull request merged**